### PR TITLE
Add message support for auth only

### PR DESCRIPTION
**Pull request by sungcheng**

### CODE REVIEW

File: src/Message/IPNCallback.php [hunk 1]
```diff
@@ -107,7 +107,7 @@ public function isCharge()
 
     /**
      * Is this a AUTH_ONLY_SINGLE_CHARGE IPN? This IPN is fired when
-     * a zero dollar transaction is made.
+     * a zero dollar transaction is made for standard subscription.
```
Comment (kaylafuchs) [inline src/Message/IPNCallback.php]: What's a standard subscription? I assumed it meant non-trial but then below you say "standard subscription with trial" - do we support any subscriptions that are not standard?

Comment (jcmanzo) [reply src/Message/IPNCallback.php]: I would mention the contract here as well. e.g. A product with a contract for a standard sub only will only emit an auth charge IPN with no subscription info.

File: src/Message/IPNCallback.php [hunk 2]
```diff
@@ -116,6 +116,18 @@ public function isAuthOnlySingleCharge()
         return $this->getIPNType() === 'AUTH_ONLY_SINGLE_CHARGE';
     }
 
+    /**
+     * Is this a AUTH_ONLY_SINGLE_CHARGE IPN? This IPN is fired when
+     * a zero dollar transaction is made for standard subscription with trial.
```
Comment (jcmanzo) [inline src/Message/IPNCallback.php]: same thing about mentioning the contract. Or better yet, maybe linking to some documentation that discusses this?

Comment (sungcheng) [reply src/Message/IPNCallback.php]: good call, let me add the url

File: src/Message/IPNCallback.php (hunk 1 above)
Comment (sungcheng) [reply src/Message/IPNCallback.php]: @kaylafuchs @jcmanzo I will include a link but basically they have multiple plan types.  The one we are using is standard subscription.  They are recommending standard subscription with trial.

**Review by jcmanzo — APPROVED** [review]

**Review by brendanpeters — APPROVED** [review]

**Pull request merged**